src/site/_collections/authors.js

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const fs = require('fs');
+const path = require('path');
+const contributors = require('../_data/contributors');
+const livePosts = require('../_filters/live-posts');
+const setdefault = require('../_utils/setdefault');
+
+/**
+ * Generate map the posts by author's username/key
+ *
+ * @param {*} posts
```

Comment:
This is an array of something? It can be `Array<*>`, that's fine too if it's a complex Eleventy type.